redis-hbo-provider/src/main/java/com/facebook/presto/statistic/PropertiesUtil.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statistic;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.google.common.collect.Maps.fromProperties;
+
+public final class PropertiesUtil
+{
+    private PropertiesUtil() {}
+
+    public static Map<String, String> loadProperties(File file)
+            throws IOException
+    {
+        Properties properties = new Properties();
+        try (InputStream in = Files.newInputStream(file.toPath())) {
+            properties.load(in);
+        }
+        return fromProperties(properties);
+    }
+
+    public static Map<String, String> loadAndProcessProperties(String path)
+            throws NoSuchFileException
+    {
+        File redisProviderFile = new File(path);
+        if (!redisProviderFile.exists()) {
+            throw new NoSuchFileException(path);
+        }
+
+        // Get Redis Properties and secrets
+        Map<String, String> properties = null;
+        try {
+            properties = new HashMap<>(loadProperties(redisProviderFile));
+            // load secrets
+            File credentialFile = new File(properties.get(RedisProviderConfig.REDIS_CREDENTIALS_PATH));
+            if (credentialFile.exists()) {
+                properties.putAll(loadProperties(credentialFile));
+            }
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Consider having it throw, rather than repackaging as `RuntimeException`.  If you're doing this because it's used in a constructor, you can move that code into a method that's annotated as `@PostConstruct`--that will ensure it's called after the constructor, but before runtime.  See examples in the code.

Reply:
On testing, I think **@PostConstruct** only applies to container-managed beans and does not work here.